go/stellar/stellarsvc/frontend.go

```diff
@@ -450,11 +483,13 @@ func (s *Server) transformPaymentRelay(ctx context.Context, acctID stellar1.Acco
 	} else {
 		loc.StatusSimplified = stellar1.PaymentStatus_CLAIMABLE
 		loc.StatusDetail = "Waiting for the recipient to open the app to claim, or the sender to cancel."
+		loc.ShowCancel = true
 	}
 	if p.Claim != nil {
 		loc.StatusSimplified = p.Claim.TxStatus.ToPaymentStatus()
 		if p.Claim.TxStatus == stellar1.TransactionStatus_SUCCESS {
 			// If the claim succeeded, the relay payment is done.
+			loc.ShowCancel = false
```

Comment:
Probably also can't claim if your current claim is in-flight. May want to show a spinner on cancel button. All that can come later though.